neet/boolean/rannetwork.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 ELIFE. All rights reserved.
+# Use of this source code is governed by a MIT
+# license that can be found in the LICENSE file.
+
+import sys
+import os
+import random as ran
+import networkx as nx
+import numpy as np
+import copy
+from .wtnetwork import WTNetwork
+
+
+
+def rewiring_fixed_degree(net):
+    '''
+    Generate a random network by rewiring a given network
+    with fixed (weighted) degree sequence, threshold and self-loops.
+
+    :param net: a network
+    :returns: a random network
+    '''
+    arr = copy.copy(net.weights)
+    length_arr = np.shape(arr)[0]
+    for i in range(length_arr):
+        for j in range(length_arr):
+            if arr[i, j] == 0:
+                continue
+            if i == j:   # to preserve self-loops
+                continue
+            edges_swiped = False
+            for k in range(length_arr):
+                if edges_swiped:
+                    break
+                for l in range(length_arr):
+                    if edges_swiped:
+                        break
+                    if k == l: # to perserve self-loops
+                        continue
+                    if arr[i, j] != arr[k, l]: # edge-swipe is allowed only between two edges sharing the same weight
+                        continue
```

Comment:
@siyuzhou Yes, I chose the rewiring under the very strict condition, which won't work for networks with continuous weights. Please go ahead and issue separate issue for it.